### PR TITLE
Feat: Update management-domains Federated Catalog docs.

### PR DESCRIPTION
**Pull request by bmg13**
## WHAT

Suggestion update for Federated Catalog usage documentation.

## WHY

To keep Federated Catalog documentation up-to-date.

## FURTHER NOTES

Management Deployments Documentation was based in this one https://github.com/eclipse-edc/Connector/blob/main/docs/developer/management-domains/management-domains.md#type-2c-catalog-servercontrol-plane-with-data-plane-runtime

Related with 
https://github.com/eclipse-tractusx/sig-release/issues/736
https://github.com/eclipse-tractusx/tractusx-edc/issues/1585

### CODE REVIEW

File: docs/development/management-domains/README.md [hunk 1]
```diff
@@ -17,14 +17,52 @@ There are several reasons why a company might consider the use of Management Dom
   different intervals or velocities. **Note that this only refers to minor changes of APIs, SPIs, configuration etc. All
   instances must still maintain protocol (DSP, DCP,...) compatibility!**
 
-For the purposes of Tractus-X, the usage of
-deployment [type 2b](https://github.com/eclipse-edc/Connector/blob/main/docs/developer/management-domains/management-domains.md#type-2b-edc-catalog-server-and-controldata-plane-runtimes)
-or [type 2c](https://github.com/eclipse-edc/Connector/blob/main/docs/developer/management-domains/management-domains.md#type-2c-catalog-servercontrol-plane-with-data-plane-runtime)
-is assumed:
 
-![type 2b](https://github.com/eclipse-edc/Connector/blob/main/docs/developer/management-domains/distributed.type2.b.svg)
+### Deployment Topologies
```
Comment (paullatzelsperger) [inline docs/development/management-domains/README.md:21]: I don't think there is much value in copy-pasting documentation content that is already present in the upstream repo, especially picture material which might be copyrighted. 

Instead, I would document how to set up a catalog server, i.e. which assets must be added there, how the resulting federated catalog looks like (flattened vs hierarchical), etc.

**Review by paullatzelsperger — CHANGES_REQUESTED** [review]

**Comment by bmg13** [comment]
@jimmarino @paullatzelsperger @ndr-brt can this PR be reopened, please?

**Comment by bmg13** [comment]
@jimmarino @paullatzelsperger @ndr-brt can this PR be reopened, please?

File: docs/development/management-domains/README.md [hunk 2]
```diff
@@ -41,7 +41,7 @@ special [asset type](https://github.com/eclipse-edc/Connector/blob/main/docs/dev
 was introduced for this purpose.
 
 Every target node produces one `Catalog`, so in the end there is a `List<Catalog>` which contains all the assets that
-are available in a particular dataspace.
+are available in a particular dataspace. Like so, after the Federated Catalog Crawler retrieves all catalogs, an aggregation of all is made and one single root catalog is exposed, similar to a Catalog Server without CatalogAssets.
```
Comment (paullatzelsperger) [inline docs/development/management-domains/README.md:44]: No, the crawler does **not** expose anything, least of all a root catalog.

**Review by paullatzelsperger — CHANGES_REQUESTED** [review]
To me, the text is quite misleading and IMO does not explain key concepts sufficiently. 

I would have expected to see at least the following aspects mentioned:
- the purpose of a catalog server, and what data structures it exposes (i.e. root catalog)
- how to set up a catalog server, i.e. that each sub-catalog is referenced via a `CatalogAsset` and how such a `CatalogAsset` looks like. This is crucial and should be accompanied by an example. For reference, you can look at [this example](https://github.com/eclipse-edc/MinimumViableDataspace/?tab=readme-ov-file#31-participants).
- the fact that querying of a catalog server is no different from querying a "normal" catalog endpoint, and what the response looks like
- the fact that contract negotiations etc. do still have to take place with the data-owning connector, not the catalog server
- the fact that the catalog server **doesn't actually** create a consolidated catalog, i.e. it does not pull back any data, it merely references other connectors' catalog endpoints
- the fact that the crawler lives on the consumer side 
- the purpose of a crawler and the fact, that it is the crawler that recursively traverses down a nested catalog 
- client-side caching: the crawler pulls back catalogs, stores them in the client-side cache (SQL), which is what the Query API uses


If I'm honest, I'm not 100% convinced that the concept of management domains and federated catalogs has been fully understood yet.

**Comment by rafaelmag110** [comment]
@bmg13 
Given the latest review from Paul, I'd suggest you to consider his points and start again from a fresh slate. I'd close this PR, spend some time understanding the concepts that are missing and then I'd get back with a new one.

**Comment by bmg13** [comment]
> @bmg13 Given the latest review from Paul, I'd suggest you to consider his points and start again from a fresh slate. I'd close this PR, spend some time understanding the concepts that are missing and then I'd get back with a new one.

Following this feedback will close this pr and revisit in the future if opportunity arises.

**Pull request closed without merging**